Packet++/src/HttpLayer.cpp

```diff
@@ -772,22 +800,16 @@ int HttpResponseFirstLine::getStatusCodeAsInt() const
 
 std::string HttpResponseFirstLine::getStatusCodeString() const
 {
-	std::string result;
-	const int statusStringOffset = 13;
 	if (!m_StatusCode.isUnsupportedCode())
 	{
-		int statusStringEndOffset = m_FirstLineEndOffset - 2;
-		if ((*(m_HttpResponse->m_Data + statusStringEndOffset)) != '\r')
-			statusStringEndOffset++;
-		result.assign((char*)(m_HttpResponse->m_Data + statusStringOffset), statusStringEndOffset-statusStringOffset);
+		return m_StatusCode.getMessage();
 	}
 
 	//else first line is illegal, return empty string
-
-	return result;
+	return "";
 }
 
-bool HttpResponseFirstLine::setStatusCode(HttpResponseStatusCode newStatusCode, std::string statusCodeString)
+bool HttpResponseFirstLine::setStatusCode(const HttpResponseStatusCode& newStatusCode)
```

Comment:
since we already support `HttpResponseStatusCode` constructor to set customized message, no need to pass message string here.
It's safe to remove the second parameter, since it's a private method.

Reply:
This method is not private, it's even used in the tests. I think we should keep it the way it is, or mark it as deprecated and add an overload

Reply:
sure, I make it back.